apps/dashboard/app/models/project.rb

```diff
@@ -28,30 +51,36 @@ def dataroot
         Pathname.new('')
       end
     end
-
-    def from_directory(project_pathname)
-      return nil unless project_pathname.directory? && project_pathname.executable? && project_pathname.readable?
-
-      Project.new({ project_directory: project_pathname.basename })
-    end
   end
 
-  validates :directory, presence: true
   validates :name, format: {
     with:    /\A[\w-]+\z/,
     message: I18n.t('dashboard.jobs_project_name_validation')
   }
 
-  attr_reader :directory
+  attr_reader :directory, :id
 
   delegate :icon, :name, :description, to: :manifest
 
   def initialize(attributes = {})
-    @directory = attributes.delete(:project_directory) || attributes[:name].to_s.downcase.tr_s(' ', '_')
-    @manifest  = Manifest.new(attributes).merge(Manifest.load(manifest_path))
+    if attributes.empty?
+      @manifest = Manifest.new({})
+    else
+      @id = attributes.delete(:id)
+      @directory = attributes.delete(:directory) || Project.dataroot.join(id.to_s)
+
+      @manifest = Manifest.new(attributes).merge(Manifest.load(manifest_path))
+    end
   end
 
   def save(attributes)
+    if id.nil? && attributes.fetch(:id, nil).nil?
+      message = 'Project ID must be set when saving.'
+      errors.add(:save, message)
+      Rails.logger.warn(message)
+      return false
```

Comment:
```suggestion
      false
```
Do we want to have `return` in the code? If so lets add it to the `lookup_table` method as well.